jwst/dark_current/tests/test_dark_sub.py

```diff
@@ -0,0 +1,517 @@
+
+    print(outfile.pixeldq[500, 500])
+    print(outfile.groupdq[0, 5, 500, 500])
+
+    # test that the NaN dark reference pixel was set to 0 (nothing subtracted)
+    assert outfile.data[0, 5, 500, 500] == 5.0
+
+    # test that the output dq file is flagged (with what)
+
+
+def test_dq_combine():
+    '''Verify that the DQ array of the dark is correctly combined with the PIXELDQ array of the science data.'''
+
+    # size of integration
+    nints = 1
+    ngroups = 5
+    xsize = 1032
+    ysize = 1024
+
+    # create raw input data for step
+    dm_ramp = make_rampmodel(nints, ngroups, ysize, xsize)
+    dm_ramp.meta.exposure.nframes = 1
+    dm_ramp.meta.exposure.groupgap = 0
+
+    # populate data array of science cube
+    for i in range(0, ngroups-1):
+        dm_ramp.data[0, i, :, :] = i
+
+    refgroups = 15
+    # create dark reference file model with fewer frames than science data
+    dark = make_darkmodel(refgroups, ysize, xsize)
+
+    # populate dq flags of sci pixeldq and reference dq
+    dm_ramp.pixeldq[500, 500] = 4
+    dm_ramp.pixeldq[500, 501] = 2
+
+    dark.dq[0, 0, 500, 500] = 1
+    dark.dq[0, 0, 500, 501] = 1
+
+    # run correction step
+    outfile = darkcorr(dm_ramp, dark)
+
+    # check that dq flags were correctly added
+    assert outfile.pixeldq[500, 500] == 5
+    assert outfile.pixeldq[500, 501] == 3
+
+
+def test_2_int():
+    '''Verify the dark correction is done by integration for MIRI observations'''
+
+    # size of integration
+    nints = 2
+    ngroups = 10
+    xsize = 1032
+    ysize = 1024
+
+    # create raw input data for step
+    dm_ramp = make_rampmodel(nints, ngroups, ysize, xsize)
+    dm_ramp.meta.exposure.nframes = 1
+    dm_ramp.meta.exposure.groupgap = 0
+
+    # populate data array of science cube
+    for i in range(0, ngroups-1):
+        dm_ramp.data[:, i, :, :] = i
+
+    refgroups = 15
+    # create dark reference file model with fewer frames than science data
+    dark = make_darkmodel(refgroups, ysize, xsize)
+
+    # populate data array of reference file
+    for i in range(0, refgroups - 1):
+        dark.data[0, i, :, :] = i * 0.1
+        dark.data[1, i, :, :] = i * 0.2
+
+    # run correction
+    outfile = darkcorr(dm_ramp, dark)
+
+    # perform subtractions manually
+
+    # check that the dark file is subtracted frame by frame from the science data
+    diff = dm_ramp.data[0, :, :, :] - dark.data[0, :ngroups, :, :]
+    diff_int2 = dm_ramp.data[1, :, :, :] - dark.data[1, :ngroups, :, :]
+
+    # test that the output data file is equal to the difference found when subtracting ref file from sci file
+
+    np.testing.assert_array_equal(outfile.data[0, :, :, :], diff,
+                                  err_msg='dark file should be subtracted from sci file ')
+    np.testing.assert_array_equal(outfile.data[1, :, :, :], diff_int2,
+                                  err_msg='dark file should be subtracted from sci file ')
+
+
+def test_dark_skipped():
```

Comment:
Isn't this effectively the same as `test_more_sci_frames` above? In which case they could be merged into a single test (just add the test of `darkstatus` to the other test).